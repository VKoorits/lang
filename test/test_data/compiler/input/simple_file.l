var a,b,c
a = "hello" + 1 + " world"
b = 3.14
c = a - b + "hello"
a = 1
