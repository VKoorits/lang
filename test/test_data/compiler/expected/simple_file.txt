CONSTANTS:
	1: hello
	2: 1
	3:  world
	4: 3.14
CODE:
1:	NEW_VAR 3
2:	PUSH 1
3:	PUSH_CONST 1
4:	PUSH_CONST 2
5:	PUSH_CONST 3
6:	BINARY_ADD
7:	BINARY_ADD
8:	STORE
9:	PUSH 2
10:	PUSH_CONST 4
11:	STORE
12:	PUSH 3
13:	PUSH 1
14:	PUSH 2
15:	PUSH_CONST 1
16:	BINARY_ADD
17:	BINARY_SUB
18:	STORE
19:	PUSH 1
20:	PUSH_CONST 2
21:	STORE
22:	DELETE_VAR 3
23:	EXIT
